### PR TITLE
Fixes #15200: Add a check_variable_match

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/15200

### CODE REVIEW

File: tree/30_generic_methods/variable_check_match_regex.cf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#  Regex must respect PCRE format.
+#
+# @parameter variable_name Complete name of the variable being tested, like my_prefix.my_variable
+# @parameter expected_match Regex to use to test if the variable content is compliant
+#
+# @class_prefix variable_check_match_regex
+# @class_parameter variable_name
+
+
+bundle agent variable_check_match_regex(variable_name, expected_match)
```
Comment (amousset) [inline tree/30_generic_methods/variable_check_match_regex.cf]: `variable_string_match` (for consistency with `condition_from_variable_match`)

**Comment by Fdall** [comment]
PR updated with a new commit

File: tree/30_generic_methods/variable_string_match.cf [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+################################################################################
+
+# @name Variable string match
+# @description Test the content of a string variable
+# @documentation Test a variable content and report a success if it matched, or an error if it does not or if the variable could not be found.
```
Comment (amousset) [inline tree/30_generic_methods/variable_string_match.cf]: Please add a mention to `condition_from_variable_match` which is useful want users only want a test and no errors.

File: tree/30_generic_methods/variable_string_match.cf [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+{
+  vars:
+      "old_class_prefix"  string => canonify("variable_string_match_${variable_name}");
+      "args"               slist => { "${variable_name}", "${expected_match}" };
+      "report_param"      string => join("_", args);
+      "full_class_prefix" string => canonify("variable_string_match_${report_param}");
+      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
+
+  classes:
+      "pass1" expression => "any";
```
Comment (amousset) [inline tree/30_generic_methods/variable_string_match.cf]: wrong order, but if it works you can probably remove it :)

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**